src/caikit_nlp_client/grpc_client.py

```diff
@@ -12,72 +10,25 @@
 log = logging.getLogger(__name__)
 
 
-@dataclass
-class GrpcConfig:
-    host: str
-    port: int
-    tls: bool = False
-    mtls: bool = False
-    ca_cert: Optional[bytes] = None
-    client_key: Optional[bytes] = None
-    client_cert: Optional[bytes] = None
-    server_cert: Optional[bytes] = None
-
-
-def make_channel(config: GrpcConfig) -> grpc.Channel:
-    log.debug(f"Making a channel from this config {config}")
-    if config.host.strip() == "":
-        raise ValueError("A non empty host name is required")
-    if config.port <= 0:
-        raise ValueError("A non zero port is required")
-
-    connection = f"{config.host}:{config.port}"
-
-    if not (config.tls or config.mtls):
-        return grpc.insecure_channel(connection)
-
-    if config.tls:
-        if config.ca_cert is None:
-            raise ValueError("A CA certificate is required")
-        return grpc.secure_channel(
-            connection,
-            grpc.ssl_channel_credentials(config.ca_cert),
-        )
-    if config.mtls:
-        if config.client_key is None:
-            raise ValueError("A client key is required")
-        if config.client_cert is None:
-            raise ValueError("A client certificate is required")
-        if config.server_cert is None:
-            raise ValueError("A server certificate is required")
-
-        return grpc.secure_channel(
-            connection,
-            grpc.ssl_channel_credentials(
-                config.server_cert, config.client_key, config.client_cert
-            ),
-        )
-    raise ValueError("invalid values")
-
-
 class GrpcClient:
     """GRPC client for a caikit nlp runtime server
 
     Args:
         channel (grpc.Channel): a connected GRPC channel for use of making the calls.
     """
 
-    def __init__(self, config: GrpcConfig) -> None:
+    def __init__(self, host: str, port: int, **kwarg) -> None:
```

Comment:
Might want to explicitly add the secure/insecure argument here, pass other CA client/server certs in kwargs

Reply:
Let me ponder on it.